operations/repo.go

```diff
@@ -203,3 +260,104 @@ func buildRepo(ctx context.Context, packages, configPath, workingDir, distro, ed
 
 	return errors.Wrapf(catcher.Resolve(), "encountered problem rebuilding repository after %d retries", retries)
 }
+
+type barqueServiceInfo struct {
+	url      string
+	username string
+	password string
+}
+
+func submitRepo(ctx context.Context, info barqueServiceInfo, configPath, distro, edition, version, arch string, packages []string) error {
+	// validate inputs
+	if edition == "community" {
```

Comment:
Should this validation be done by RespositoryConfig instead of the client?

Reply:
I"m mostly trying to avoid incompatibilities between this code and the previous operation for the local mode version.